### DISCUSSION TITLE
Filter Related Models Using The $relations Array not working

### DISCUSSION THREAD

**Discussion by fhsdmhacc** [Tucker-Eric/EloquentFilter#192] [Q&A] [answered]
// Model Company
/**
     * Get the user associated with the company.
     */
    public function user(): HasOne
    {
        return $this->hasOne(User::class);
    }

// Company Filter
class CompanyFilter extends ModelFilter
{
    public $relations = [
        'user' => [
            'search'
        ]
    ];

    public function search($search)
    {
        return $this->when($search, function ($query) use ($search) {
            $query->where('name', 'like', "%$search%")
                ->orWhere('address', 'like', "%$search%");
        });
    }
}

// User Filter
class UserFilter extends ModelFilter
{
    public $relations = [];

    public function search($search)
    {
        return $this->when($search, function ($query) use ($search) {
            $query->where('name', 'like', "%$search%")
                ->orWhere('email', 'like', "%$search%");
        });
    }
}

// Controller
Company::with('user')->filter(request()->all())->paginate(5);

// SQL Check from debugbar
select * from `companies` where (`name` like '%ful%' or `address` like '%ful%' and exists (select * from `users` where `companies`.`id` = `users`.`company_id` and (`name` like '%ful%' or `email` like '%ful%'))) and 0 = 1 order by `id` desc limit 6 offset 0

// Question
When I enter this url http://127.0.0.1:8000/companies?search=ful no data appears, even though the data is available. Thanks for reply

**Comment by Tucker-Eric** [comment, marked as answer]
Looks like `and 0 = 1` in your query is appended somehow from outside the filters. That is what looks like your issue is. This doesn't look like a filter issue. I'd check your scopes on that model to try to identify where that is being appended.
